spark/src/main/scala/org/apache/spark/sql/delta/PreprocessTableWithDVs.scala

```diff
@@ -93,20 +91,22 @@ object ScanWithDeletionVectors {
 
     // See if the relation is already modified to include DV reads as part of
     // a previous invocation of this rule on this table
-    if (fileFormat.hasDeletionVectorMap) return None
+    if (fileFormat.hasBroadcastHadoopConf) return None
```

Comment:
We should update that check since using a broadcast conf isn't a marker of the preprocess step for DVs (we actually remove this need in this change..)

We could instead check if the scan already contains the `IS_ROW_DELETED_STRUCT_FIELD` field, e.g.:


```
    if (scan.output.exists(_.name == IS_ROW_DELETED_COLUMN_NAME)) return None
```